cpp/include/cudf/io/orc.hpp

```diff
@@ -649,7 +649,7 @@ class orc_writer_options {
    *
    * @param meta Associated metadata
    */
-  void set_metadata(table_input_metadata const* meta) { _metadata = meta; }
+  void set_metadata(table_input_metadata meta) { _metadata = std::move(meta); }
```

Comment:
Maybe using `&&` is better since we can avoid copying if not needed.
```suggestion
  void set_metadata(table_input_metadata&& meta) { _metadata = std::move(meta); }
```